### PR TITLE
- fixed metrics if you use a table prefix

### PR DESCRIPTION
**Pull request by karaktaka**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 node_modules
 phpunit.xml
 vendor
+/.idea/
```
Comment (GrahamCampbell) [inline .gitignore]: Please do not add this.

File: app/Repositories/Metric/AbstractMetricRepository.php [hunk 2]
```diff
@@ -58,6 +58,11 @@ protected function getTableName()
         $connection = $this->config->get('database.connections.'.$driver);
         $prefix = $connection['prefix'];
 
-        return $prefix.'metrics';
+        $return = (object) array(
```
Comment (GrahamCampbell) [inline app/Repositories/Metric/AbstractMetricRepository.php]: Please update the phpdoc. Also, an array would be fine here.

Comment (GrahamCampbell) [inline app/Repositories/Metric/AbstractMetricRepository.php]: Infact, it's better to just make two methods here than returning something like this.

File: .gitignore (hunk 1 above)
Comment (jbrooksuk) [inline .gitignore]: We should make a CS check that these don't get changed 😂

File: app/Repositories/Metric/AbstractMetricRepository.php (hunk 2 above)
Comment (jbrooksuk) [inline app/Repositories/Metric/AbstractMetricRepository.php]: Also, we use the new array syntax only.

Comment (karaktaka) [inline app/Repositories/Metric/AbstractMetricRepository.php]: Feel free to modify it. It was just a suggestion.

File: app/Repositories/Metric/AbstractMetricRepository.php [hunk 3]
```diff
@@ -58,6 +58,11 @@ protected function getTableName()
...
+                        'metrics_base' => $prefix.'metrics',
```
Comment (ConnorVG) [inline app/Repositories/Metric/AbstractMetricRepository.php]: I'm not a fan of concatenation. Especially when we have possible PHP7 optimisations we can have here.

**Comment by ConnorVG** [comment]
Doesn't it make more sense to just do something like this? https://gist.github.com/ConnorVG/51c863929ce0f3c6800c0a4aa16b13f8

**Pull request closed without merging**